### ISSUE TITLE
`register_tmp_handler` does not appear to be temporary

### ISSUE THREAD

**Issue by bklang** [adhearsion/adhearsion#541]
Code:

```Ruby
      agent.call.register_tmp_handler :event, Punchblock::Event::Unjoined do
        agent.callback :disconnect, self, agent.call, queued_call
        ignoring_ended_calls { queued_call.hangup }
        ignoring_ended_calls { conditionally_return_agent agent if agent.call.active? }
        agent.call[:queued_call] = nil
      end
```

The code within the `:disconnect` callback is what attempts to add the caller back to the queue.

Logs (increases by one line each time the agent takes a call, which corresponds to that callback being added):

```
[2015-02-20 17:38:38.042] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.046] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.049] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.051] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.054] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.056] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.059] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.061] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.063] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.065] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.067] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.070] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.072] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.075] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.077] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.079] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.081] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.083] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.086] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.088] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.090] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.092] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.095] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.097] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.099] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.101] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.104] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.106] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.108] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.111] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.114] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.116] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.119] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.121] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.123] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.126] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
[2015-02-20 17:38:38.128] DEBUG ElectricSlide::CallQueue: Not returning agent #<Agent: extension: 1001, campaign_ref: 2044, state: wrapup, call: a6546d41-90a7-4b9b-9864-cf5ba0dc347a> to the queue
```

**Comment by victorluft** [comment]
See https://github.com/adhearsion/has-guarded-handlers/pull/2

**Comment by benlangfeld** [comment]
This should be fixed in has-guarded-handlers 1.6.2. Please test and re-open if it persists.
